tests/unit/src/test/scala/tests/worksheets/WorksheetSuite.scala

```diff
@@ -233,9 +240,62 @@ class WorksheetSuite extends FunSuite {
       |val y = "foobar".stripSuffix("bar")
       |""".stripMargin,
     """|
-       |<val x = "foobar".stripSuffix("bar")> // "foo"
+       |<val x = "foobar".stripSuffix("bar")> // "foo": String
        |x: String = "foo"
        |""".stripMargin
   )
 
+  checkDecorations(
+    "types-rendering",
+    """
+      |val a = 1
+      |val b: String = "A"
+      |val (x, y) = (true, 0)
+      |a + 4
+      |class C {}
+      |def f(): String = "FFF"
+      |f()
+      |
+      |def px(): String = { println("Something"); "OK" }
+      |
+      |println("print")
+      |
+      |px()
+      |
+      |{ println("P1"); List(1,2,3) }
+      |""".stripMargin,
+    """|
+       |<val a = 1> // 1: Int
+       |<val b: String = "A"> // "A": String
+       |<val (x, y) = (true, 0)> // x: Boolean=true, y: Int=0
+       |<a + 4> // 5: Int
+       |class C {}
+       |def f(): String = "FFF"
+       |<f()> // "FFF": String
+       |
+       |def px(): String = { println("Something"); "OK" }
+       |
+       |<println("print")> // print
+       |
+       |<px()> // "OK": String
+       |
+       |<{ println("P1"); List(1,2,3) }> // List(1, 2, 3): List[Int]
+       |""".stripMargin,
+    screenWidth = 80,
+    details = false
+  )
+
+  checkDecorations(
+    "types-too-long",
+    """|
+       |val a = Map(List(3) -> Map("v" -> Set(true, false)))
+       |val b = a
+      """.stripMargin,
+    """|
+       |<val a = Map(List(3) -> Map("v" -> Set(true, false)))> // Map(List(3) -> Map("v...
```

Comment:
Would it make sense to try and display the type at the beginning so that it doesn't get truncated for large values?

Reply:
We could try the following formatting

```
<val a = Map(List(3) -> Map("v" -> Set(true, false)))> //: Map[List[Int], Set[Boolean]] = Map...
```

I don't have very strong opinions on what is the best solution here. cc/ @tpolecat any suggestions on how to format types in worksheets when the value is too large and gets truncated to fit the print width?